### PR TITLE
Get the upstream remote dynamically

### PR DESCRIPTION
**Pull request by xiangpengzhao**
Fixes: #380

Note: not fully tested yet...

/cc @david-mcmahon @wojtek-t @ixdy

### CODE REVIEW

File: lib/gitlib.sh [hunk 1]
```diff
@@ -262,3 +262,13 @@ gitlib::repo_state () {
     return 1
   fi
 }
+
+##############################################################################
+# Returns the name of the upstream remote repository name for the local git
+# repo, e.g. "upstream" or "origin".
+#
+gitlib::upstream_remote_name() {
```
Comment (xiangpengzhao) [inline lib/gitlib.sh]: Yeah I steal this which @ixdy mentioned here https://github.com/kubernetes/release/issues/380#issuecomment-315874982 :)

**Comment by david-mcmahon** [comment]
Note that relnotes is soon to be replaced by #434.

**Comment by xiangpengzhao** [comment]
@david-mcmahon yeah I reviewed that PR today. Since I don't know when that PR will be landing, I just sent this PR for relnotes in case we still need the script. Feel free to close it if it's indeed not necessary :)

File: relnotes [hunk 2]
```diff
@@ -325,7 +327,7 @@ generate_notes () {
                    "Specify on the command-line. Exiting..."
   fi
 
-  range="$start_tag..$release_tag"
+  range="$start_tag..$upstream_remote_name/$release_tag"
```
Comment (david-mcmahon) [inline relnotes]: Why is this necessary?

Comment (xiangpengzhao) [reply relnotes]: TBH, I'm not so sure about this. I saw @wojtek-t used this as a workaround in https://github.com/kubernetes/release/issues/380
```
-  range="$start_tag..${release_tag:-$branch_head}"
+  range="$start_tag..upstream/${release_tag:-$branch_head}"
```

**Comment by ixdy** [comment]
do we still need this PR? or should we close?

**Comment by xiangpengzhao** [comment]
Closing it would be ok for now. Let's reopen it when we find we do need it.
/close
